creator-node/src/snapbackSM/peerSetManager.js

```diff
@@ -193,14 +203,58 @@ class PeerSetManager {
    * @param {string} endpoint
    * @returns {Object} the /health_check/verbose response
    */
-  async determinePeerHealth (endpoint) {
+  async getPeerHealth (endpoint) {
     // Axios request will throw on timeout or non-200 response
-    return axios({
+    const resp = await axios({
       baseURL: endpoint,
       url: '/health_check/verbose',
       method: 'get',
       timeout: PEER_HEALTH_CHECK_REQUEST_TIMEOUT
     })
+
+    return resp.data.data
+  }
+
+  /**
+   * Takes data off the verbose health check response and determines the peer heatlh
+   * @param {Object} verboseHealthCheckResp verbose health check response
+   *
+   * TODO: consolidate CreatorNodeSelection + peer set health check calculation logic
+   */
+  determinePeerHealth (verboseHealthCheckResp) {
+    const { storagePathSize, storagePathUsed } = verboseHealthCheckResp
+    if (storagePathSize && storagePathUsed && storagePathSize - storagePathUsed <= MINIMUM_STORAGE_PATH_SIZE) {
+      throw new Error(`Almost out of storage=${storagePathSize - storagePathUsed}bytes remaining`)
+    }
+
+    // Check for sufficient memory space of 6GB
+    const { usedMemory, totalMemory } = verboseHealthCheckResp
+    if (usedMemory && totalMemory && totalMemory - usedMemory <= MINIMUM_MEMORY_UNUSED) {
+      throw new Error(`Running low on memory=${totalMemory - usedMemory}bytes remaining`)
+    }
+
+    // Check for sufficient file descriptors space of minimum 5%
+    const { allocatedFileDescriptors, maxFileDescriptors } = verboseHealthCheckResp
+    if (allocatedFileDescriptors && maxFileDescriptors && allocatedFileDescriptors / maxFileDescriptors >= MAX_FILE_DESCRIPTORS_OPEN) {
```

Comment:
seems like `MAX_FILE_DESCRIPTERS_OPEN` actually should be called `MAX_FILE_DESCRIPTORS_ALLOCATED` ?